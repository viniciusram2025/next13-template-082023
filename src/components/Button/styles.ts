import styled, { css } from 'styled-components';

export const Wrapper = styled.button`
  ${({ theme }) => css`
    padding: ${theme.spacings.small} ${theme.spacings.medium};
  `}
`;
