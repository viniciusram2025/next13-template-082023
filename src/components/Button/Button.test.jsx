import { describe, it, expect } from '@jest/globals';
import { screen } from '@testing-library/react';

import { Button } from '.';
import { renderTheme } from '../../styles/renderTheme';

describe('Button', () => {
  it('should render the component', () => {
    const { container } = renderTheme(<Button>DefaultButton</Button>);

    const button = screen.getByRole('button', { name: 'DefaultButton' });

    expect(button).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
