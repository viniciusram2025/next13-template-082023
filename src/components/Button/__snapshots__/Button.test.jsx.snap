// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render the component 1`] = `
.c0 {
  padding: 1.6rem 2.4rem;
}

<div>
  <button
    class="c0"
  >
    DefaultButton
  </button>
</div>
`;
